### PR TITLE
several test enhancements

### PR DESCRIPTION
**Pull request by pawelru**
A few enhancements including:
- consistent usage of `grDevices::pdf(nullfile())` replacing `grDevices::pdf()` - a follow-up on https://github.com/insightsengineering/teal.widgets/pull/203
- better catch errors in the test `"type_download_srv_table: pagination, lpp to small"`
- DRY for `plot_types` variable
- remove `expect_download`. This is not _only_ expectation that the downloadable link exists but it also downloads the file and run snapshot testing on it. The problem is that this file is named from sys time on the moment of test run (e.g. `tests/testthat/_snaps/app_driver_pws_ui/pws-001-plot_20240805_034140.png`) which creates a new file every time you run the test (that's why our integration test fails). Moreover, I believe that this is not the intend of that unit test to test regression of a dummy plot.

### CODE REVIEW

File: tests/testthat/test-table_with_settings.R [hunk 1]
```diff
@@ -110,20 +110,17 @@ testthat::test_that("type_download_srv_table: downloading different output types
...
-          session$setInputs(`pagination_switch` = TRUE)
-          session$setInputs(`lpp` = 1)
-          session$setInputs(`file_format` = down_type)
-          testthat::expect_true(file.exists(output$data_download))
-        }
+  shiny::testServer(
+    teal.widgets:::type_download_srv_table,
+    args = list(id = "tws", table_reactive = table_r),
+    expr = {
+      for (down_type in c(".txt", ".png")) {
```
Comment (donyunardi) [inline tests/testthat/test-table_with_settings.R:117]: Any reason why we're checking `.png` here when the ui doesn't offer `.png` option?
https://github.com/insightsengineering/teal.widgets/blob/main/R/table_with_settings.R#L144

Instead, should we replace `.png` with `.pdf`, which we do offer?

**Review by donyunardi — APPROVED** [review]
Just had a small comment about checking `.png` output. 
Other than that everything else looks good.

File: tests/testthat/test-table_with_settings.R (hunk 1 above)
Comment (pawelru) [reply tests/testthat/test-table_with_settings.R:117]: whoops my bad
thanks for the spot!

Comment (pawelru) [reply tests/testthat/test-table_with_settings.R:117]:
```suggestion
      for (down_type in c(".txt", ".pdf")) {
```

**Pull request merged**